### PR TITLE
Implemented CreateWorld

### PR DESCRIPTION
**Pull request by PeachesMLG**
# Description
State the changes of the pull request here.

# Checklist
The following items should be checked before the pull request can be merged.
- [ ] Version number updated in `settings.gradle`.
- [ ] Unit tests added.
- [ ] Code follows existing code format.

# Info on creating a pull request
Before the pull request can be accepted, the version number stored in settings.gradle should be updated.
MockBukkit uses semantic-versioning (https://semver.org/)

 - If the PR fixes a bug in MockBukkit, update the patch version. (if the current version is `0.4.1`, the new version should be `0.4.2`)
 - If the PR adds a new feature to MockBukkit, update minor version. (if the current version is `0.4.1`, the new version should be `0.5.0`)
 - Make sure that unit tests are added which test the relevant changes.
 - Make sure that the changes follow the existing code format.

Note that a PR that fixes an `UnimplementedOperationException` should be considered a new version and not a bugfix.

The version number can be found in `settings.gradle`.

### CODE REVIEW

File: src/main/java/be/seeseemelk/mockbukkit/ServerMock.java
Comment (TheBusyBiscuit) [inline src/main/java/be/seeseemelk/mockbukkit/ServerMock.java]: Invalid implementations, please follow our specifications and throw an `UnimplementedOperationException` for non-implemented method stubs.

File: src/main/java/be/seeseemelk/mockbukkit/WorldMock.java
Comment (TheBusyBiscuit) [inline src/main/java/be/seeseemelk/mockbukkit/WorldMock.java]: Please avoid wildcard imports

File: src/main/java/be/seeseemelk/mockbukkit/entity/PlayerMock.java
Comment (TheBusyBiscuit) [inline src/main/java/be/seeseemelk/mockbukkit/entity/PlayerMock.java]: Invalid implementations, please follow our specifications and throw an `UnimplementedOperationException` for non-implemented method stubs.

File: src/test/java/be/seeseemelk/mockbukkit/ServerMockTest.java
Comment (TheBusyBiscuit) [inline src/test/java/be/seeseemelk/mockbukkit/ServerMockTest.java]: Please avoid wildcard imports

File: src/test/java/be/seeseemelk/mockbukkit/ServerMockTest.java
Comment (TheBusyBiscuit) [inline src/test/java/be/seeseemelk/mockbukkit/ServerMockTest.java]: `Server#createWorld` returns the created World.
It would be much more sensible to store it in a var and use that for your assertions instead of `server.getWorlds().get(0)`

**Review by TheBusyBiscuit — CHANGES_REQUESTED** [review]

File: src/test/java/be/seeseemelk/mockbukkit/ServerMockTest.java
Comment (PeachesMLG) [reply src/test/java/be/seeseemelk/mockbukkit/ServerMockTest.java]: Thats true, good point

File: src/main/java/be/seeseemelk/mockbukkit/WorldMock.java
Comment (PeachesMLG) [reply src/main/java/be/seeseemelk/mockbukkit/WorldMock.java]: Not quite sure why you dont like wildcard imports, they look way nicer, but whatever. Took me ages to find out how to make my ide disable them. Your code formatting conventions are annoying af

File: src/main/java/be/seeseemelk/mockbukkit/WorldMock.java
Comment (TheBusyBiscuit) [inline src/main/java/be/seeseemelk/mockbukkit/WorldMock.java]:
```suggestion
	public WorldMock(@NotNull WorldCreator creator)
	{
```

**Review by TheBusyBiscuit — CHANGES_REQUESTED** [review]
One small change left (which you can just commit right from my review), then it is good to go!

File: src/main/java/be/seeseemelk/mockbukkit/WorldMock.java
Comment (TheBusyBiscuit) [reply src/main/java/be/seeseemelk/mockbukkit/WorldMock.java]: > they look way nicer, but whatever

Wildcard imports are not a style choice, they are a functional utility.
Their purpose is to import entire namespaces which can reduce the file size when you got many imports from the same namespace. However, the 90s are over, so we don't need to worry about every single bit or byte in our file anymore.

Nowadays, wildcard imports are sometimes regarded as a bad practice as they lead to so-called namespace pollution.
By nature, wildcard imports will automatically import any classes from the given namespace into the current class.
Overuse of this can lead to conflicts and collisions ([Wikipedia](https://en.wikipedia.org/wiki/Name_collision)), as the whole point of a namespace is to seperate class names which are non-unique.
By using a wildcard import, you are opening the door to ambiguities and potential collisions.

Check this for more info and a pretty good example:
https://rules.sonarsource.com/java/RSPEC-2208

And as you can guess, the `org.bukkit` namespace is by no means a small one...

Comment (PeachesMLG) [reply src/main/java/be/seeseemelk/mockbukkit/WorldMock.java]: 99% sure it gets optimized on compilation, but whatever

Comment (TheBusyBiscuit) [reply src/main/java/be/seeseemelk/mockbukkit/WorldMock.java]: This is not a runtime issue, it's a bad development practice and not happening on the compiler level.
But either way, I gave you a decent explanation, what you do with this information and advice is all your call.

**Review by TheBusyBiscuit — APPROVED** [review]
LGTM

**Pull request merged**
